crates/lunatic-process/src/runtimes/wasmtime.rs

```diff
@@ -155,15 +155,7 @@ where
                 Err(err) => {
                     // If the trap is a result of calling `proc_exit(0)`, treat it as an no-error finish.
                     match err.downcast_ref::<wasmtime::Trap>() {
-                        Some(trap) => {
-                            if trap.i32_exit_status().is_some()
-                                && trap.i32_exit_status().unwrap() == 0
-                            {
-                                ResultValue::Ok
-                            } else {
-                                ResultValue::Failed(trap.to_string())
-                            }
-                        }
+                        Some(trap) => ResultValue::Failed(trap.to_string()),
```

Comment:
You will still need to check for an exit code `0`, but instead of using `wasmtime:Trap` you will need to use: https://docs.rs/wasmtime-wasi/3.0.1/wasmtime_wasi/struct.I32Exit.html#impl-Pointee-for-I32Exit

Reply:
I updated this based on the conversation in discord.